ADDING_NEW_RULES.md

```diff
@@ -0,0 +1,94 @@
+* Implement new rule using the process below
+
+For the below example, let's look at implementing a new rule that [TODO]
+
+If you want to take a look at a complete set of changes that implement this new rule before diving into the instructions, see [TODO this commit on Github](https://github.com/MobilityData/gtfs-validator).
+
+### 1. Add the rule to [`RULES.md`](RULES.md)
+
+Add the rule to the error or warnings table at the top of [`RULES.md`](RULES.md):
+
+~~~
+
+~~~
+
+...and add a definition of that rule at the bottom of the errors or warnings section:
+
+~~~
+
+~~~
+
+### 2. Add a new `*Notice.java` class for the new rule output
+
+All classes that implements rules output are under the [`notice` package](https://github.com/MobilityData/gtfs-validator/tree/master/domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice) of the domain layer
+
+They must extend either `ErrorNotice` or `WarningNotice` and fit the `*Notice` format
+
+Start by adding the error code definition to the corresponding `ErrorNotice` or `WarningNotice` class located in the `base` package, for example
+
+`protected static final String E_0XX = "E0XX";`
+
+You can see here in the constructor signature what fields your own Notice class will inherit
+
+~~~
+public ErrorNotice(String filename,
+                   String noticeId,
+                   String title,
+                   String description,
+                   String entityId)
+~~~
+
+Add your new class in the `error` or `warning` package accordingly
+
+You can define fields and getters that are not already covered by the base classes
+
+Your new class must override and define an `export` method like the following
+
+    @Override
+    public void export(final NoticeExporter exporter) throws IOException {
+        exporter.export(this);
+    }
+
+### 3. Implement your new rule exportation code
+
+Add a new definition at the bottom of the [`NoticeExporter` interface](https://github.com/MobilityData/gtfs-validator/blob/master/domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/NoticeExporter.java)
+
+Add the required code to the interface implementations in the [`exporter` package](https://github.com/MobilityData/gtfs-validator/tree/master/adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter)
+
+**Note**: JSON export required at a minimum. Leaving the protobuf implementation empty will not lead to your PR being rejected.
+
+### 4. Implement exportation code unit tests
+
+Add a new test implementation to the [`*ExporterTest` classes](https://github.com/MobilityData/gtfs-validator/tree/master/adapter/exporter/src/test/java/org/mobilitydata/gtfsvalidator/exporter)
+
+### 5. Implement your new rule in a specific use case
+
+All rules in the validator are implemented in self-contained classes in the [`usecase` package](https://github.com/MobilityData/gtfs-validator/tree/master/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase)
```

Comment:
I suggest adding a note that the `usecase` package isn't in the domain module, but is in the `usecase` module.